### PR TITLE
add breaking-changes section to changelog

### PR DESCRIPTION
**Pull request by bastelfreak**
Hi,
this PR allows us to create another section in the changelog, for breaking changes. This is not a 100% implementation of https://github.com/skywinder/github-changelog-generator/issues/316, but may help people. Tested this successfully on https://github.com/voxpupuli/puppet-dhcp/blob/master/CHANGELOG.md

### CODE REVIEW

File: lib/github_changelog_generator/parser.rb [hunk 1]
```diff
@@ -129,6 +132,9 @@ def self.setup_parser(options)
         opts.on("--enhancement-labels  x,y,z", Array, 'Issues with the specified labels will be always added to "Implemented enhancements" section. Default is \'enhancement,Enhancement\'') do |list|
           options[:enhancement_labels] = list
         end
+        opts.on("--breacking-labels x,y,z", Array, 'Issues with these lables will be added to a new section, called "Breaking Changes". Default is \'backwards-incompatible\'') do |list|
```
Comment (olleolleolle) [inline lib/github_changelog_generator/parser.rb]: Typos:

- `breacking-labels` => `breaking-labels` 
- `lables` => `labels`

**Comment by olleolleolle** [comment]
Looking at the generated CHANGELOG.md, I feel that it'd be cool to place any breaking changes section FIRST.

**Comment by olleolleolle** [comment]
So, there's room for careful refactoring - or disabling a RuboCop check to pass this:

```
lib/github_changelog_generator/generator/generator.rb:105:5: C: Metrics/CyclomaticComplexity: Cyclomatic complexity for parse_by_sections is too high. [8/7]
    def parse_by_sections(issues, pull_requests)
    ^^^

```

**Comment by bastelfreak** [comment]
I tried to refactor it, but didn't find a clean solution. If anybody has ideas please let me know. I will try to place the section at the beginning.

**Comment by olleolleolle** [comment]
@bastelfreak Failing forward is also a good option!

Add a `# rubocop:disable Metrics/CyclomaticComplexity` as a comment on the line of the `def ` of the offending method; and the crypt keepers will get to this gnarly bit at some point in the future. 

Don't worry about it.

**Comment by bastelfreak** [comment]
@olleolleolle I reordered the output, the breaking stuff should now be the first.

**Comment by bastelfreak** [comment]
Hi @olleolleolle, what needs to be done to get this merged?

**Comment by bastelfreak** [comment]
I made some tests again and I'm not 100% if it still works after I changed the ordering. I will have a look tomorrow.

**Comment by hunner** [comment]
> Q: If a PR is marked as both Breaking and an Enhancement, it would be in both lists, right?

No, backwards-incompatible (or "breaking") changes are always features because you are enhancing the project with behavior that is desirable, so it would not make sense to put them in both sections.

For a standardized workflow around this, see http://keepachangelog.com/ . It's great, because it tries to make a portable standardized changelog format that works anywhere, and has a formal definition for what goes in each section and how to adhere to http://semver.org/ .

**Comment by bastelfreak** [comment]
Hi @olleolleolle  @skywinder could you recheck this PR?
I fixed one issue and rebased against latest master.

**Comment by bastelfreak** [comment]
Hi @olleolleolle, can you check this PR as well? Would be awesome!

**Comment by olleolleolle** [comment]
Feedback right away" Avoid using `push`, use "the shovel operator" (`<<`) to append to lists.

**Comment by bastelfreak** [comment]
* replaced push with shovel operator
* rebased

File: lib/github_changelog_generator/generator/generator.rb [hunk 2]
```diff
@@ -96,60 +96,74 @@ def create_log_for_tag(pull_requests, issues, newer_tag, older_tag = nil)
...
 
+    # rubocop:disable  Metrics/CyclomaticComplexity
     # This method sort issues by types
     # (bugs, features, or just closed issues) by labels
     #
     # @param [Array] issues
     # @param [Array] pull_requests
-    # @return [Array] tuple of filtered arrays: (Bugs, Enhancements Issues)
+    # @return [Array] tuple of filtered arrays: (Bugs, Enhancements, Breaking stuff,  Issues)
     def parse_by_sections(issues, pull_requests)
```
[the remark this answers is not in the document]
Comment (alexjfisher) [reply lib/github_changelog_generator/generator/generator.rb]: At the moment, the breaking change appears in multiple sections.  See https://github.com/voxpupuli/puppet-rabbitmq/blob/33bc9ef7f69155d602f1cdd5abf5a794d1d8a27c/CHANGELOG.md#v700-2017-09-14 for example of where this branch was recently used.
@hunner has suggested this actually might not be the best behaviour. https://github.com/skywinder/github-changelog-generator/pull/530#issuecomment-321611067

[the remark this answers is not in the document]
Comment (ekohl) [reply lib/github_changelog_generator/generator/generator.rb]: https://github.com/ekohl/github-changelog-generator/commit/f5cb492ee6d835c11846002e00796a798e9b90df is my attempt at a cleanup and has tests. I kept current code if multiple labels match but maybe breaking should be at the top.

**Comment by bastelfreak** [comment]
@olleolleolle I added the commit from @ekohl, could you take a look again?

**Comment by olleolleolle** [comment]
@bastelfreak Thanks for that. 

Could you fix the lint complaints?

**Comment by olleolleolle** [comment]
@bastelfreak Thanks for passing lint checks, I can merge this from here.

**Comment by nfischer** [comment]
Should this be documented on the README?

**Comment by pme123** [comment]
I could not find, how to use this. 

I tried the label `backwards-incompatible` - but without success.

**Pull request merged**